packages/react-devtools-shared/src/__tests__/profilingCache-test.js

```diff
@@ -367,87 +388,173 @@ describe('ProfilingCache', () => {
-          <Context2.Provider value={true}>
-            <Component count={2} />
-          </Context2.Provider>
+          <Component count={2} />
         </Context.Provider>,
         container,
       ),
     );
 
-    // Third render has a changed reducer hook
+    // Third render has a changed reducer hook.
     utils.act(() => dispatch({type: 'invert'}));
 
-    // Fourth render has a changed state hook
+    // Fourth render has a changed state hook.
     utils.act(() => setState('def'));
 
-    // Fifth render has a changed context value for context 1, but no changed hook.
+    // Fifth render has a changed context value, but no changed hook.
     utils.act(() =>
       legacyRender(
         <Context.Provider value={false}>
-          <Context2.Provider value={true}>
-            <Component count={2} />
-          </Context2.Provider>
+          <Component count={2} />
         </Context.Provider>,
         container,
       ),
     );
 
-    // Sixth render has another changed context value for context 2, but no changed hook.
-    utils.act(() =>
-      legacyRender(
-        <Context.Provider value={false}>
-          <Context2.Provider value={false}>
-            <Component count={2} />
-          </Context2.Provider>
-        </Context.Provider>,
-        container,
-      ),
-    );
+    // 6th renderer is triggered by a sync external store change.
+    utils.act(() => {
+      snapshot++;
+      syncExternalStoreCallback();
+    });
+
     utils.act(() => store.profilerStore.stopProfiling());
 
-    const allCommitData = [];
+    const rootID = store.roots[0];
 
-    function Validator({commitIndex, previousCommitDetails, rootID}) {
-      const commitData = store.profilerStore.getCommitData(rootID, commitIndex);
-      if (previousCommitDetails != null) {
-        expect(commitData).toEqual(previousCommitDetails);
-      } else {
-        allCommitData.push(commitData);
-        expect(commitData).toMatchSnapshot(
-          `CommitDetails commitIndex: ${commitIndex}`,
+    const allChangeDescriptions = [];
+
+    function getChangeDescriptions(commitIndex, label) {
+      let changeDescriptions;
+
+      function Validator() {
+        const commitData = store.profilerStore.getCommitData(
+          rootID,
+          commitIndex,
         );
-      }
-      return null;
-    }
 
-    const rootID = store.roots[0];
+        changeDescriptions = commitData.changeDescriptions;
+
+        allChangeDescriptions.push(changeDescriptions);
+
+        return null;
+      }
 
-    for (let commitIndex = 0; commitIndex < 6; commitIndex++) {
       utils.act(() => {
-        TestRenderer.create(
-          <Validator
-            commitIndex={commitIndex}
-            previousCommitDetails={null}
-            rootID={rootID}
-          />,
-        );
+        TestRenderer.create(<Validator />);
       });
+
+      return changeDescriptions;
     }
 
-    expect(allCommitData).toHaveLength(6);
+    // 1st render: No change
```

Comment:
I pulled these bits of the snapshot data inline so it's more clear what they're testing. (They fail on main if the same test is run.)